### PR TITLE
feat: add font-style selector.

### PR DESCRIPTION
**Pull request by tanujvyas10**
- Add font-picker component.
-Add myfont.scss
-Update shared.module.ts
-Update settings.component.html
-Update toolbar.component.html
-Update main.scss

## Description
Now user can select the different font style for the Single Page Application.
## Related issues and discussion
#472

## Screenshots, if any
ezgif com-video-to-gif

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by tanujvyas10** [comment]
@abhaychawla Please review my work.

File: src/app/shared/font-picker/font-picker.component.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#font-selector {
+    max-width: 4rem;
+   }
```
Comment (abhaychawla) [inline src/app/shared/font-picker/font-picker.component.scss]: There shouldn't be so much space between the font selector and notifications icon. Please correct.

Comment (abhaychawla) [inline src/app/shared/font-picker/font-picker.component.scss]: There shouldn't be so much space between the font selector and notifications icon. Please correct.

File: src/app/shared/font-picker/font-picker.component.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+
```
Comment (abhaychawla) [inline src/app/shared/font-picker/font-picker.component.scss]: Every file should only end with a single line at the end.

File: src/app/core/shell/toolbar/toolbar.component.html [hunk 3]
```diff
@@ -60,7 +60,9 @@
   <mifosx-language-selector class="ml-1 language" fxHide.lt-md></mifosx-language-selector>
 
   <mifosx-theme-picker fxHide.lt-md></mifosx-theme-picker>
-
+  
+
```
Comment (abhaychawla) [inline src/app/core/shell/toolbar/toolbar.component.html]: Add only a single line for space not two.

File: src/app/core/shell/toolbar/toolbar.component.html [hunk 4]
```diff
@@ -60,7 +60,9 @@
   <mifosx-language-selector class="ml-1 language" fxHide.lt-md></mifosx-language-selector>
 
   <mifosx-theme-picker fxHide.lt-md></mifosx-theme-picker>
-
+  
+  
+  <mifosx-font-picker></mifosx-font-picker>
```
Comment (abhaychawla) [inline src/app/core/shell/toolbar/toolbar.component.html]: Should be hidden in smaller screens. Update accordingly.

File: src/app/core/shell/toolbar/toolbar.component.html [hunk 5]
```diff
@@ -60,7 +60,9 @@
...
   <!--
```
Comment (abhaychawla) [inline src/app/core/shell/toolbar/toolbar.component.html]: Add a single line for space.

File: src/app/settings/settings.component.html [hunk 6]
```diff
@@ -90,15 +90,9 @@
...
-          <mat-label>Default Font</mat-label>
-          <mat-select>
-            <mat-option *ngFor="let font of fonts" [value]="font">
-              {{ font }}
-            </mat-option>
-          </mat-select>
-        </mat-form-field>
+       
+        <span class="header">Font style</span>
+      <mifosx-font-picker></mifosx-font-picker>
```
Comment (abhaychawla) [inline src/app/settings/settings.component.html]: Indent properly.

File: src/app/shared/font-picker/font-picker.component.html [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<button mat-icon-button [matMenuTriggerFor]="fontMenu" matTooltip="Font Styles">
+ <h2>Aa</h2>
+</button>
+
+<mat-menu  #fontMenu="matMenu" x-position="before">
```
Comment (abhaychawla) [inline src/app/shared/font-picker/font-picker.component.html]:
```suggestion
<mat-menu #fontMenu="matMenu" x-position="before">
```

File: src/app/shared/font-picker/font-picker.component.html [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+      <mat-option (click)="onFontChange('my-first-font')" value="my-first-font">Monospace</mat-option>
+      <mat-option (click)="onFontChange('my-second-font')" value="my-second-font">Anton</mat-option>
+      <mat-option (click)="onFontChange('my-third-font')" value="my-third-font">Roboto</mat-option>
+</mat-menu>
```
Comment (abhaychawla) [inline src/app/shared/font-picker/font-picker.component.html]: Use for loop to add options for fonts using an object. They should also be properly renamed instead of `my-font`. Ensure they are in alphabetical order.

File: src/app/shared/font-picker/font-picker.component.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit} from '@angular/core';
```
Comment (abhaychawla) [inline src/app/shared/font-picker/font-picker.component.ts]: Add all relevant comments to the file.

File: src/app/shared/font-picker/font-picker.component.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+   }
+   /**
+    *     * Function for chaning the currentfont to the new font the user choose.
+    */
+  onFontChange(font: string) {
+    document.body.classList.remove(this.currentFont);
+    document.body.classList.add(font);
+    localStorage.setItem('mifosXfont', font);
+    this.currentFont = font;
+  }
```
Comment (abhaychawla) [inline src/app/shared/font-picker/font-picker.component.ts]: Please shift the logic into services similar to what has been done for theme.

File: src/app/shared/shared.module.ts [hunk 11]
```diff
@@ -1,51 +1,53 @@
  /** Angular Imports */
```
Comment (abhaychawla) [inline src/app/shared/shared.module.ts]: All unnecessary changes related to indentation should be removed from you commit.

File: src/main.scss [hunk 12]
```diff
@@ -10,7 +10,9 @@
...
 
-
 // 3rd party libraries
 $fa-font-path: "../node_modules/font-awesome/fonts";
-@import "../node_modules/font-awesome/scss/font-awesome.scss";
\ No newline at end of file
+@import "../node_modules/font-awesome/scss/font-awesome.scss";
+
+//import the Font for MifosX
+@import './fonts/myfont'
```
Comment (abhaychawla) [inline src/main.scss]: Use proper name.

File: src/main.scss [hunk 13]
```diff
@@ -10,7 +10,9 @@
...
 @import "assets/styles/helper";
 
-
 // 3rd party libraries
 $fa-font-path: "../node_modules/font-awesome/fonts";
-@import "../node_modules/font-awesome/scss/font-awesome.scss";
\ No newline at end of file
+@import "../node_modules/font-awesome/scss/font-awesome.scss";
+
+//import the Font for MifosX
```
Comment (abhaychawla) [inline src/main.scss]: Comment properly similar to what has been done for theme.

File: src/fonts/myfont.scss [hunk 14]
```diff
@@ -0,0 +1,36 @@
+@import '~@angular/material/theming';
```
Comment (abhaychawla) [inline src/fonts/myfont.scss]: Separate all fonts into their files. Ensure they are properly renamed. Comments and variable names should be also updated to reflect the same.

**Comment by abhaychawla** [comment]
@tanujvyas10 Are you able to update the PR?

**Comment by abhaychawla** [comment]
Closing due to long inactivity.
